# PENGERTIAN MATRIKS#
Matrix adalah kumpulan data atau elemen yang terletak pada posisi tertentu berdasarkan baris dan kolomnya. Matrix banyak digunakan dalam bidang seperti matematika, ilmu komputer, fisika, dan ekonomi untuk menunjukkan berbagai macam struktur data atau data.
* macam-macam matriks, diantaranya:

1. Matriks baris-kolom, juga dikenal sebagai matriks biasa, adalah jenis matriks yang paling umum, di mana elemennya diatur dalam baris dan kolom.

2. Matrik Nol: Matriks di mana semua elemennya nol. Ini biasanya ditulis sebagai 0 0 atau 0 × 0 m×n untuk matriks × m×n.
3. Matriks Identitas: Matriks persegi dengan semua elemen diagonal utama adalah 1 dan semua elemen diagonal lainnya adalah 0. Matriks ini biasanya ditulis sebagai I atau I n untuk matriks identitas berukuran × n × n.
4. Matriks diagonal adalah matriks di mana semua elemen tidak ada di luar diagonal utama.
5. Matriks invers: Matriks yang menghasilkan matriks identitas jika dikalikan dengan matriks aslinya.
6. Matriks singuler adalah matriks di mana tidak ada determinannya.
7. matriks persegi dengan semua elemen di atas diagonal utamanya adalah nol.
* operasi matriks
1. penjumlahan
2. pengurangan
3. perkalian
4. transpose matriks








In [ ]:
# Mendefinisikan matriks A
A = [[2, 1, 3],
     [2, 3, 6],
     [4, 2, 8]]

# Mendefinisikan matriks B
B = [[2, 3, 6],
     [4, 2, 9],
     [3, 6, 8]]

# Mendefinisikan matriks C
C = [[2, 8],
     [4, 7],
     [5, 7]]


**1. matriks penjumplahan**

Syarat penjumlaha matriks adalah bahwa jika dua matriks, seperti matriks A dan B, memiliki ordo yang sama, maka elemen-elemen yang seletak dapat dijumlahkan. Jumlah matriks A dan B dapat ditulis dengan rumus A+B,
Berikut adalah contoh rumus penjumlahan matriks.

matriks A:
$$
matriks A=\begin{bmatrix}
a & b & c \\
d & e & f
\end{bmatrix}
$$
matriks B:
$$
matriks B=\begin{bmatrix}
g & h & i \\
j & k & l \\
\end{bmatrix}
$$
A+B=
$$
A+B =\begin{bmatrix}
a+g & b+h & c+i \\
d+j & e+k & f+l \\
\end{bmatrix}
$$




In [ ]:
def penjumlahan_matriks(matriks1, matriks2):
    hasil_penjumlahan = []
    for i in range(len(matriks1)):
        baris_hasil = []
        for j in range(len(matriks1[0])):
            baris_hasil.append(matriks1[i][j] + matriks2[i][j])
        hasil_penjumlahan.append(baris_hasil)
    return hasil_penjumlahan

In [ ]:
#  Contoh Penjumlahan matriks A dan B
hasil_penjumlahan = penjumlahan_matriks(A, B)
print("Hasil Penjumlahan A + B:")
for row in hasil_penjumlahan:
    print(row)
print()

Hasil Penjumlahan A + B:
[4, 4, 9]
[6, 5, 15]
[7, 8, 16]



Matrix A dan B dapat dijumlahkan dengan menjumlahkan elemen-elemen yang sesuai dari kedua matrix, karena keduanya memiliki dimensi yang sama, yaitu 3×3 3×3. Untuk menambah kedua matrix tersebut, ikuti langkah-langkah berikut:


matrix A:

$$
matriks A=\begin{bmatrix}
2 & 1 & 3 \\
2 & 3 & 6 \\
4 & 2 & 8
\end{bmatrix}
$$

matrix B:

$$
matriks B=\begin{bmatrix}
2 & 3 & 6 \\
4 & 2 & 9 \\
3 & 6 & 8
\end{bmatrix}
$$

Untuk menjumlahkan matrix A + B, maka kita harus menjumlahkan setiap elemen yang sesuai:

$$
A + B = \begin{bmatrix}
2+2 & 1+3 & 3+6 \\
2+4 & 3+2 & 6+9 \\
4+3 & 2+6 & 8+8
\end{bmatrix}
$$

$$
A + B = \begin{bmatrix}
4 & 4 & 9 \\
6 & 5 & 15 \\
7 & 8 & 16
\end{bmatrix}
$$

Jadi, hasil dari penjumlahan matrix A + B adalah:

$$
\begin{bmatrix}
4 & 4 & 9 \\
6 & 5 & 15 \\
7 & 8 & 16
\end{bmatrix}
$$


**2.PENGURANGAN MATRIKS**

Syarat pengurangan matriks yaitu : jika terdapat dua matriks, misal matriks A dan B, yang memiliki ordo sama, maka elemen-elemen yang seletak dapat dikurangkan. selisih matriks A dan matriks B dapat dinyatakan dengan A – B.
matriks A:
$$
matriks A=\begin{bmatrix}
a & b & c \\
d & e & f
\end{bmatrix}
$$
matriks B:
$$
matriks B=\begin{bmatrix}
g & h & i \\
j & k & l \\
\end{bmatrix}
$$
A-B =
$$
A-B =\begin{bmatrix}
a-g & b-h & c-i \\
d-j & e-k & f-l \\
\end{bmatrix}
$$



In [ ]:
def pengurangan_matriks(matriks1, matriks2):
    hasil_pengurangan = []
    for i in range(len(matriks1)):
        baris_hasil = []
        for j in range(len(matriks1[0])):
            baris_hasil.append(matriks1[i][j] - matriks2[i][j])
        hasil_pengurangan.append(baris_hasil)
    return hasil_pengurangan

In [ ]:
# Pengurangan matriks A dan B
hasil_pengurangan = pengurangan_matriks(A, B)
print("Hasil Pengurangan A - B:")
for row in hasil_pengurangan:
    print(row)
print()

Hasil Pengurangan A - B:
[0, -2, -3]
[-2, 1, -3]
[1, -4, 0]



Untuk melakukan pengurangan matriks \( A - B \), kita mengurangkan setiap elemen matriks \( B \) dari elemen yang sesuai di matriks \( A \).

Matrix \( A \) dan \( B \) yang telah disebutkan sebelumnya yaitu:

$$
A = \begin{bmatrix} 2 & 1 & 3 \\ 2 & 3 & 6 \\ 4 & 2 & 8 \end{bmatrix}
$$

$$
B = \begin{bmatrix} 2 & 3 & 6 \\ 4 & 2 & 9 \\ 3 & 6 & 8 \end{bmatrix}
$$

Langkah-langkah pengurangan \( A - B \) adalah sebagai berikut:

$$
A - B = \begin{bmatrix} 2-2 & 1-3 & 3-6 \\ 2-4 & 3-2 & 6-9 \\ 4-3 & 2-6 & 8-8 \end{bmatrix}
$$

Hasilnya adalah:

$$
A - B = \begin{bmatrix} 0 & -2 & -3 \\ -2 & 1 & -3 \\ 1 & -4 & 0 \end{bmatrix}
$$

Jadi, hasil pengurangan matriks \( A - B \) adalah:

$$
A - B = \begin{bmatrix} 0 & -2 & -3 \\ -2 & 1 & -3 \\ 1 & -4 & 0 \end{bmatrix}
$$


**3. PERKALIAN MATRIKS**

Perkalian dua matriks memiliki aturan yang berbeda dari perkalian skalar, yang hanya mengalikan setiap elemen matriks dengan bilangan skalar. Sebuah matriks, misalnya matriks A dan matriks B, dapat dikalikan jika banyaknya kolom matriks A sama dengan banyaknya baris matriks B.

 matriks $A$ dan $B$:

$$
A = \begin{bmatrix} a & b & c \\ d & e & f \\ g & h & i \end{bmatrix}
$$

$$
B = \begin{bmatrix} j & k & l \\ m & n & o \\ p & q & r \end{bmatrix}
$$

Langkah-langkah perkalian matriks $A$ dengan $B$ tetap sama yaitu:

$$
A \cdot B = \begin{bmatrix} aj + bm + cp & ak + bn + cq & al + bo + cr \\
                            dj + em + fp & dk + en + fq & dl + eo + fr \\
                            gj + hm + ip & gk + hn + iq & gl + ho + ir \end{bmatrix}
$$

Jadi, hasil perkalian matriks $A$ dengan $B$ dalam bentuk variabel adalah:

$$
A \cdot B = \begin{bmatrix} aj + bm + cp & ak + bn + cq & al + bo + cr \\
                            dj + em + fp & dk + en + fq & dl + eo + fr \\
                            gj + hm + ip & gk + hn + iq & gl + ho + ir \end{bmatrix}
$$


In [ ]:
def perkalian_matriks(matriks1, matriks2):
    hasil_perkalian = []
    for i in range(len(matriks1)):
        baris_hasil = []
        for j in range(len(matriks2[0])):
            total = 0
            for k in range(len(matriks1[0])):
                total += matriks1[i][k] * matriks2[k][j]
            baris_hasil.append(total)
        hasil_perkalian.append(baris_hasil)
    return hasil_perkalian

In [ ]:
#Perkalian matriks A dan B
hasil_perkalian = perkalian_matriks(A, B)
print("Hasil Perkalian A * B:")
for row in hasil_perkalian:
    print(row)
print()


Hasil Perkalian A * B:
[17, 26, 45]
[34, 48, 87]
[40, 64, 106]



Aturan perkalian matriks digunakan untuk melakukan perkalian matriks \(A) dengan \(B). Dalam metode ini, setiap elemen baris pertama dari matriks pertama dikalikan dengan setiap elemen kolom pertama dari matriks kedua, dan kemudian hasilnya ditambahkan. Prosedur ini dilakukan untuk setiap elemen baris dan kolom.


Matrix \(A\) dan \(B\) yang telah disebutkan sebelumnya adalah:

$$
A = \begin{bmatrix} 2 & 1 & 3 \\ 2 & 3 & 6 \\ 4 & 2 & 8 \end{bmatrix}
$$

$$
B = \begin{bmatrix} 2 & 3 & 6 \\ 4 & 2 & 9 \\ 3 & 6 & 8 \end{bmatrix}
$$

Langkah-langkah perkalian matriks \(A\) dengan \(B\) adalah sebagai berikut:

$$
A \cdot B = \begin{bmatrix} 2\cdot2+1\cdot4+3\cdot3 & 2\cdot3+1\cdot2+3\cdot6 & 2\cdot6+1\cdot9+3\cdot8 \\
                            2\cdot2+3\cdot4+6\cdot3 & 2\cdot3+3\cdot2+6\cdot6 & 2\cdot6+3\cdot9+6\cdot8 \\
                            4\cdot2+2\cdot4+8\cdot3 & 4\cdot3+2\cdot2+8\cdot6 & 4\cdot6+2\cdot9+8\cdot8 \end{bmatrix}
$$

Hasilnya adalah:

$$
A \cdot B = \begin{bmatrix} 2+4+9 & 6+2+18 & 12+9+24 \\
                            4+12+18 & 6+6+36 & 12+27+48 \\
                            8+8+24 & 12+4+48 & 24+18+64 \end{bmatrix}
$$

$$
A \cdot B = \begin{bmatrix} 17 & 26 & 45 \\
                            34 & 48 & 87 \\
                            40 & 64 & 106 \end{bmatrix}
$$

Jadi, hasil perkalian matriks \(A\) dengan \(B\) adalah:

$$
A \cdot B = \begin{bmatrix} 17 & 26 & 45 \\
                            34 & 48 & 87 \\
                            40 & 64 & 106 \end{bmatrix}
$$


**4. INVERSE MATRIKS**

contoh: perkalian matriks z=A.(B.C)

In [ ]:
BC = perkalian_matriks(B,C)
Z = perkalian_matriks(A,BC)
for row in Z:
    print(row)
print()

[363, 633]
[695, 1217]
[866, 1510]



*Untuk operasi A.(B.C)*

Langkah pertama adalah melakukan perkalian matriks \( B \) dengan matriks \( C \) untuk mendapatkan hasil \( BC \). Prosesnya adalah sebagai berikut:

Matriks B:

$$
B = \begin{bmatrix} 2 & 3 & 6 \\ 4 & 2 & 9 \\ 3 & 6 & 8 \end{bmatrix}
$$

Matriks C :
$$
C = \begin{bmatrix} 2 & 8 \\ 4 & 7 \\ 5 & 7 \end{bmatrix}
$$

Perkalian B dengan C akan menghasilkan matriks BC dengan nilai:
$$
BC = \begin{bmatrix} 50 & 77 \\ 69 & 121 \\ 74 & 115 \end{bmatrix}
$$

Langkah 2: Perkalian Matriks A dengan BC
Langkah kedua adalah mengalikan matriks \( A \) dengan hasil perkalian \( BC \) untuk mendapatkan matriks \( Z \). berikut prosesnya:

$$
\text{Matriks } A :
$$

$$
A = \begin{bmatrix} 2 & 1 & 3 \\ 2 & 3 & 6 \\ 4 & 2 & 8 \end{bmatrix}
$$

$$
\text{Hasil perkalian } BC :
$$

$$
BC = \begin{bmatrix} 50 & 77 \\ 69 & 121 \\ 74 & 115 \end{bmatrix}
$$

Perkalian A dengan BC akan menghasilkan matriks Z dengan nilai:
$$
Z = \begin{bmatrix} 363 & 633 \\ 695 & 1217 \\ 866 & 1510 \end{bmatrix}
$$

$$
\text{Ini adalah hasil dari perkalian matriks yang telah dihitung menggunakan kode Python yang diberikan sebelumnya.}
$$


**5. TRANSPOSE MATRIKS**

Transpose matriks adalah matriks baru yang elemen baris dan kolomnya dikembalikan ke elemen baris dan kolom matriks sebelumnya. Dengan kata lain, transpose matriks dibentuk oleh pembalikan elemen baris dan kolom menjadi elemen baris dan kolom matriks sebelumnya.

$$
Z = \begin{bmatrix} a11 & b \\ c & d \\ e & f \end{bmatrix}
$$


Transpose dari Z \text disimbolkan dengan Z^T akan mengubah baris menjadi kolom dan sebaliknya, sehingga:

$$
Z^T = \begin{bmatrix} a & c & 5 \\ b & d & 7 \end{bmatrix}
$$


In [ ]:
def transpose_matriks(matriks):
    # Mendapatkan jumlah baris dan kolom matriks
    baris = len(matriks)
    kolom = len(matriks[0])

    # Membuat matriks baru dengan ukuran kolom x baris
    transposed_matriks = [[0 for _ in range(baris)] for _ in range(kolom)]

    # Melakukan transposisi matriks
    for i in range(baris):
        for j in range(kolom):
            transposed_matriks[j][i] = matriks[i][j]

    return transposed_matriks

In [ ]:
# Mendapatkan transpose dari matriks C
transpose_C = transpose_matriks(C)
# Cetak hasil transpose
print("Transpose dari matriks C:")
for row in transpose_C:
    print(row)

Transpose dari matriks C:
[2, 4, 5]
[8, 7, 7]


Untuk melakukan transpose pada matriks C langkah-langkahnya adalah sebagai berikut:

matriks C
$$
C = \begin{bmatrix} 2 & 8 \\ 4 & 7 \\ 5 & 7 \end{bmatrix}
$$
transpose dari C, disimbolkan dengan CT, akan mengubah baris menjadi kolom dan sebaliknya, sehingga:

$$
C^T = \begin{bmatrix} 2 & 4 & 5 \\ 8 & 7 & 7 \end{bmatrix}
$$

Ini adalah matriks hasil transpose dari matriks C. Baris pertama dari matriks asal masuk ke kolom pertama, baris kedua masuk ke kolom kedua, dan seterusnya.

